### PR TITLE
SWDEV-409068 - Fix paths with spaces in it

### PR DESCRIPTION
**Pull request by kzhuravl**

### CODE REVIEW

**Comment by kzhuravl** [comment]
Ping

**Comment by kzhuravl** [comment]
Ping

File: bin/hipcc.pl [hunk 1]
```diff
@@ -469,10 +466,7 @@ BEGIN
     if (not $isWindows and $escapeArg) {
         $arg =~ s/[^-a-zA-Z0-9_=+,.\/]/\\$&/g;
     }
-    if ($isWindows and $escapeArg) {
-        $arg =~ s/[^-a-zA-Z0-9_=+,.:\/\\]/\\$&/g;
-    }
-    $toolArgs .= " $arg" unless $swallowArg;
+    $toolArgs .= " \"$arg\"" unless $swallowArg;
```
Comment (yxsamliu) [inline bin/hipcc.pl:469]: I saw a use case that --rocm-path="c:/hip build" is passed to hipcc. In that case, we may not want to quote the arg.

**Review by david-salinas — COMMENTED** [review]
I see a lot of the changes are not specific to Windows (not in if($isWindows)).  Should we keep these changes to be Windows specific?

Also, would it be cleaner to, just escape all of the path strings when we're done building the $HIPCC string array to be used in CMD ?  This was the intent of my original patch.

File: bin/hipcc.pl (hunk 1 above)
Comment (kzhuravl) [reply bin/hipcc.pl:469]: @yxsamliu, I stripped all the quotes from all the strings. So this is needed.

**Comment by kzhuravl** [comment]
> I see a lot of the changes are not specific to Windows (not in if($isWindows)). Should we keep these changes to be Windows specific?
> 
> Also, would it be cleaner to, just escape all of the path strings when we're done building the $HIPCC string array to be used in CMD ? This was the intent of my original patch.

We talked about this offline. I don't think it is worth doing anything significant to hipcc perl scripts at this point as I want to concentrate on hipcc.bin and deprecate perl scripts.

File: bin/hipcc.pl (hunk 1 above)
Comment (yxsamliu) [reply bin/hipcc.pl:469]: that should work

**Review by yxsamliu — APPROVED** [review]
LGTM

**Pull request merged**